### PR TITLE
DoS attack possible when reading from CRLDistributionPoints

### PR DESCRIPTION
**Pull request by saadk10**
In cfssl/revoke/revoke.go, there are multiple places which use ioutil.ReadAll to read the CRL. If a rogue certificate is sent with a CRLDistributionPoint to http://mal-server/100G.file, then this code will consume all memory.

Added setter functions that allow you to pass in a function that handles how to read the http response body for various HTTP Gets allowing for more flexibility.

### CODE REVIEW

**Comment by lziest** [comment]
Thanks, will review soon, lgtm at first glance

File: revoke/revoke.go [hunk 1]
```diff
@@ -303,3 +304,24 @@ func sendOCSPRequest(server string, req []byte, leaf, issuer *x509.Certificate)
 
 	return ocsp.ParseResponseForCert(body, leaf, issuer)
 }
+
+var crlRead = ioutil.ReadAll
+
+// SetCRLFetcher sets the function to use to read from the http response body
```
Comment (lziest) [inline revoke/revoke.go]: I hope we can mention 'DDoS' to justify it.

**Comment by smithbk** [comment]
Yes, without this, there seems to be no way to avoid a DoS attack, so I think it is very important.
I think there are 3 options:
1) Allow CFSSL users to provide their own read function as this pull request does.
2) CFSSL provides its own read function which limits the read buffer to some configurable max size.  This would make it easier for all users of CFSSL.
3) Both 1 & 2, but given #2 I can't think of why someone would need #1 but you never know.

**Comment by saadk10** [comment]
@lziest Any updates on this?

**Comment by lziest** [comment]
LGTM

**Pull request merged**